java/common/src/main/scala/org/typelevel/otel4s/java/TextMapPropagatorImpl.scala

```diff
@@ -39,4 +39,18 @@ private[java] class TextMapPropagatorImpl[F[_]: Sync](
     Sync[F].delay(
       jPropagator.inject(toJContext(ctx), carrier, fromTextMapSetter)
     )
+
+  def injected[A](ctx: Vault, carrier: A)(implicit
+      injector: TextMapUpdater[A]
+  ): A = {
```

Comment:
why does `inject` need `Sync#delay` but `injected` doesn't?

Reply:
`inject` is mutating the carrier and `injected` is returning a new, immutable carrier.